backend/clj/rehearser/api/exercise.clj

```diff
@@ -1,5 +1,30 @@
 (ns rehearser.api.exercise
-  (:require [rehearser.service.exercise :as service]))
+  (:require [rehearser.service.exercise :as service]
+            [clojure.spec.alpha :as s]))
+
+(s/def ::id int?)
+(s/def ::account-id int?)
+(s/def ::title string?)
+(s/def ::description string?)
+
+(s/def ::metadata
+  (s/keys :req-un [::id ::account-id]))
+
+(s/def ::exercise-content
+  (s/keys :req-un [::title
+                   ::description]))
+(s/def ::exercise
+  (s/merge ::exercise-content
+           ::metadata))
+
+(s/def ::exercise-update
```

Comment:
There is not much in the way of processing these. Means I end up repeating myself quite a lot. This leads to somewhat more verbose specs.